frontend/app/utils/open-payments.server.ts

```diff
@@ -1,20 +1,85 @@
 import {
-  type AuthenticatedClient,
   type PendingGrant,
   type WalletAddress,
-  createAuthenticatedClient,
+  type AuthenticatedClient,
   isFinalizedGrant,
-  isPendingGrant
+  isPendingGrant,
+  createAuthenticatedClient
 } from '@interledger/open-payments'
 import { createId } from '@paralleldrive/cuid2'
 import { toWalletAddressUrl } from './utils.server'
+import { createContentDigestHeader } from 'httpbis-digest-headers'
+import { signMessage } from 'http-message-signatures/lib/httpbis'
+import type { Request } from 'http-message-signatures'
+import * as ed from '@noble/ed25519'
+import { Header } from '~/components'
+
+interface RequestLike extends Request {
+  body?: string
+}
+
+interface SignOptions {
+  request: RequestLike
+  privateKey: Uint8Array
+  keyId: string
+}
+
+export interface SignatureHeaders {
+  'Signature': string
+  'Signature-Input': string
+}
+
+interface ContentHeaders {
+  'Content-Digest': string
+  'Content-Length': string
+  'Content-Type': string
+}
+type Headers = SignatureHeaders & Partial<ContentHeaders>
 
+/**
+ * Creates an authenticated Open Payments client for making signed requests.
+ * Based on the Interledger Web Monetization Extension implementation:
+ * @see https://github.com/interledger/web-monetization-extension/blob/main/src/background/services/openPayments.ts#L163
+ */
 async function createClient(env: Env) {
```

Comment:
Since this runs in CF Workers, maybe we can initialize the client as a singleton, and we would not create a new client when a worker is reused?

i.e.:

```ts
let client: AuthenticatedClient;

async function createClient() {
  if(client) return client
  
  return await createAuthenticatedClient({ ... })
}
```